rllib/rl_trainers/torch_rl_distributed_trainer.py

```diff
@@ -0,0 +1,115 @@
+import numpy as np
+import ray
+import torch
+import ray.train as train
+from ray.air import session
+
+from ray.air.config import ScalingConfig
+
+from torch.nn.parallel import DistributedDataParallel as DDP
+
+from ray.train._internal.utils import construct_train_func
+from ray.train._internal.backend_executor import BackendExecutor
+from ray.train.torch import TorchConfig
+from ray.train import TrainingIterator
+from ray.train._internal.checkpoint import CheckpointManager
+from ray.train._internal.dataset_spec import RayDatasetSpec
+from ray.util.queue import Queue
+
+
+class TorchRLDataParallelTrainer:
```

Comment:
Some high-level interface stuff to make this consistent with the current design of RLTrainer, This is the abstraction that is expected from an RLTrainer:
```
@abc.abstractmethod
	update(batch: BatchType, **kwargs) -> Any → Performs an update and returns anything that might be needed inside the algorithm.

```

In multi-gpu case you pass in a batch of data and it should automatically  distribute that batch into sub-batches in the Queue object that was created earlier. The trainer actors will update the model as soon as the queue get's field. But who enqueues the data into those queues? trainer.update(batch) should do this.